WhisperAPI/WhisperAPI.Tests/Unit/SuggestionsServiceTest.cs

```diff
@@ -45,12 +47,35 @@ public void When_receive_valid_searchresult_from_search_then_return_list_of_sugg
                 .Setup(x => x.GetNlpAnalysis(It.IsAny<string>()))
                 .Returns(nlpAnalysis);
 
-            this._suggestionsService.GetSuggestions(querry).Should().BeEquivalentTo(this.GetSuggestedDocuments());
+            this._suggestionsService.GetSuggestions(this.GetSearchQueries()).Should().BeEquivalentTo(this.GetSuggestedDocuments());
         }
 
         [Test]
-        [TestCase("test")]
-        public void When_receive_empty_searchresult_from_search_then_return_empty_list_of_suggestedDocuments(string querry)
+        [TestCase]
+        public void When_receive_valid_searchresult_from_search_then_return_list_of_suggestedDocuments_filter_sent_suggestion()
+        {
+            var intents = new List<Intent>
+            {
+                new IntentBuilder().WithName("Need Help").Build()
+            };
+
+            var nlpAnalysis = new NlpAnalysisBuilder().WithIntents(intents).Build();
```

Comment:
This won't work because you use the last query to decide whether you use all previous queries or none of them.

This method is getting too complex. We should have a better separation of concerns. Something like:

public bool IsQueryRelevant(SearchQuery query)
        {
            // Do NLP analysis of query
            // return if relevant
        }
        
        public void UpdateContext(ConversationContext context, SearchQuery query)
        {
            // query.Relevant = IsQueryRelevant(query)
            // Add query to context.SearchQueries
        }

And finally, the method would be reduced to:
public IEnumerable<SuggestedDocument> GetSuggestions(ConversationContext context)
        {
            // Make a long query by joining only the relevant ones (.where(q => q.Relevant))
            // Search Coveo Index with this long query
            // Filter out already used documents
            // return
        }

What do you think ?

Reply:
Did you place this comment in the wrong file ?